sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample04_BasicCheckpointing.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Processor.Samples.Infrastructure;
+using Azure.Storage.Blobs;
+
+namespace Azure.Messaging.EventHubs.Processor.Samples
+{
+    /// <summary>
+    ///   An introduction to the Event Processor client, illustrating how to create simple checkpoints.
+    /// </summary>
+    ///
+    public class Sample04_BasicCheckpointing : IEventHubsBlobCheckpointSample
+    {
+        /// <summary>
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name => nameof(Sample04_BasicCheckpointing);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description => "An introduction to the Event Processor client, illustrating how to how to create simple checkpoints.";
+
+        /// <summary>
+        ///   Runs the sample using the specified Event Hubs and Azure storage connection information.
+        /// </summary>
+        ///
+        /// <param name="eventHubsConnectionString">The connection string for the Event Hubs namespace that the sample should target.</param>
+        /// <param name="eventHubName">The name of the Event Hub, sometimes known as its path, that the sample should run against.</param>
+        /// <param name="blobStorageConnectionString">The connection string for the storage account where checkpoints and state should be persisted.</param>
+        /// <param name="blobContainerName">The name of the blob storage container where checkpoints and state should be persisted.</param>
+        ///
+        public async Task RunAsync(string eventHubsConnectionString,
+                                   string eventHubName,
+                                   string blobStorageConnectionString,
+                                   string blobContainerName)
+        {
+            // A checkpoint is a the term used to describe a snapshot of the state of processing for a partition which has been persisted
```

Comment:
It's almost like you can see my thought process as I revised things.... since I seem to be not quite cleaning up prior phrasing all over the place.   /sigh